File: packages/python/qsu/os/getSystemUptime.py
import math
from typing import Optional, Union

from ..format.numberFormat import numberFormat
from ._system import systemUptime


def getSystemUptime(opt: Optional[dict] = None) -> Union[int, float, str]:
	t = systemUptime()

	if not t:
		return 0

	if opt and opt.get('floor'):
		t = math.floor(t)

	return numberFormat(t) if opt and opt.get('format') else t

File: packages/javascript/lib/node/os/getSystemUptime.ts
import { uptime } from 'node:os';
import { numberFormat } from '../../format/numberFormat.js';
import type { GetUptimeOption } from '../../_types/global.js';

export function getSystemUptime(opt?: GetUptimeOption): number | string {
	let t = uptime();

	if (!t) {
		return 0;
	}

	if (opt?.floor) {
		t = Math.floor(t);
	}

	return opt?.format ? numberFormat(t) : t;
}
